demos/glacial_isostatic_adjustment/gia_2d_cylindrical/gia_2d_cylindrical.py

```diff
@@ -0,0 +1,550 @@
+# Let's get started!
+# The first step is to import the gadopt module, which
+# provides access to Firedrake and associated functionality. We also import pyvista and matplotlib, for plotting.
+
+from gadopt import *
+from gadopt.utility import step_func, vertical_component
+import pyvista as pv
+import matplotlib.pyplot as plt
+
+
+# In this tutorial we are going to use a fully unstructured mesh, created using [Gmsh](https://gmsh.info/). We have used Gmsh to construct a triangular mesh with a target resolution of 200km near the surface of the Earth coarsening to 500 km in the interior. Gmsh is a widely used open source software for creating finite element meshes and Firedrake has inbuilt functionaly to read Gmsh '.msh' files. Take a look at the *annulus_unstructured.py* script in this folder which creates the mesh using Gmsh's Python api.
+#
+# As this problem is not formulated in a Cartesian geometry we set the `mesh.cartesian`
+# attribute to `False`. This ensures the correct configuration of a radially inward vertical direction.
+
+# Set up geometry:
+from unstructured_annulus import generate_mesh
+generate_mesh()
+mesh = Mesh("unstructured_annulus_refined_surface.msh")
+bottom_id, top_id = 1, 2
+mesh.cartesian = False
+D = 2891e3  # Depth of domain in m
+
+# We next set up the function spaces, and specify functions to hold our solutions. As our mesh is now made up of triangles instead of quadrilaterals, the syntax for defining our finite elements changes slighty. We need to specify *Continuous Galerkin* elements, i.e. replace `Q` with `CG` instead.
+
+# +
+# Set up function spaces - currently using the bilinear Q2Q1 element pair:
+V = VectorFunctionSpace(mesh, "CG", 2)  # (Incremental) Displacement function space (vector)
+W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+TP1 = TensorFunctionSpace(mesh, "DG", 2)  # (Discontinuous) Stress tensor function space (tensor)
+R = FunctionSpace(mesh, "R", 0)  # Real function space (for constants)
+
+Z = MixedFunctionSpace([V, W])  # Mixed function space.
+
+z = Function(Z)  # A field over the mixed function space Z.
+u, p = split(z)  # Returns symbolic UFL expression for u and p
+z.subfunctions[0].rename("Incremental Displacement")
+z.subfunctions[1].rename("Pressure")
+
+displacement = Function(V, name="displacement").assign(0)
+stress_old = Function(TP1, name="stress_old").assign(0)
+# -
+
+# We can output function space information, for example the number of degrees
+# of freedom (DOF).
+
+# Output function space information:
+log("Number of Incremental Displacement DOF:", V.dim())
+log("Number of Pressure DOF:", W.dim())
+log("Number of Velocity and Pressure DOF:", V.dim()+W.dim())
+
+# We can now visualise the resulting mesh. As you can see there is finer resolutation near the surface compared with the lower mantle.
+
+VTKFile("mesh.pvd").write(Function(V))
+mesh_data = pv.read("mesh/mesh_0.vtu")
+edges = mesh_data.extract_all_edges()
+plotter = pv.Plotter(notebook=True)
```

Comment:
There is some graphical stuff here that needs to be gated with 
```
# + tags=["active-ipynb"]
```

Reply:
Thanks!